src/main/java/wolox/training/models/User.java

```diff
@@ -59,23 +60,26 @@ public String getUsername() {
     }
 
     public void setUsername(String username) {
-        username = username;
+        Preconditions.checkNotNull(username);
```

Comment:
After checking if a field is null, check if the field is empty.

An error message can be added to each check :

```
Preconditions.checkNotNull(username, "The username field cannot be null.");
Preconditions.checkArgument(! username.isEmpty(), "The username field cannot be empty.");
this. username = username;
```

I recommend using a class to store error messages as constants.

Reply:
I put the messages and added it like constants.